scripts/variantstore/wdl/GvsImportGenomes.wdl

```diff
@@ -16,19 +16,43 @@ workflow GvsImportGenomes {
     Boolean skip_loading_vqsr_fields = false
 
     File interval_list = "gs://gcp-public-data--broad-references/hg38/v0/wgs_calling_regions.hg38.noCentromeres.noTelomeres.interval_list"
-    # If increasing this, also consider increasing `load_data_preemptible_override` and `load_data_maxretries_override`.
-    Int load_data_batch_size = 5
+    Int? load_data_batch_size
     Int? load_data_preemptible_override
     Int? load_data_maxretries_override
     File? load_data_gatk_override = "gs://broad-dsp-spec-ops/scratch/bigquery-jointcalling/jars/gg_VS-443_VETIngestValidation_20220531/gatk-package-4.2.0.0-531-gf8f4ede-SNAPSHOT-local.jar"
     String? service_account_json_path
   }
 
+  Int num_samples = length(external_sample_names)
+  Int max_auto_batch_size = 20000
+
+  if ((num_samples > max_auto_batch_size) && !(defined(load_data_batch_size))) {
+    call Utils.TerminateWorkflow as DieDueToTooManySamplesWithoutExplicitLoadDataBatchSize {
+      input:
+        message = "Importing " + num_samples + " samples but `load_data_batch_size` not explicitly specified; limit for auto batch-sizing is " + max_auto_batch_size + " samples."
+    }
+  }
+
+  # At least 1, per limits above not more than 20.
+  Int effective_load_data_batch_size = if (defined(load_data_batch_size)) then select_first([load_data_batch_size])
+                                       else if num_samples < 1000 then 1
+                                            else num_samples / 1000
+
+  # At least 3, per limits above not more than 5.
+  Int effective_load_data_preemptible = if (defined(load_data_preemptible_override)) then select_first([load_data_preemptible_override])
+                                        else if effective_load_data_batch_size < 12 then 3
+                                             else effective_load_data_batch_size / 4
+
+  # At least 3, per limits above not more than 5.
+  Int effective_load_data_maxretries = if (defined(load_data_maxretries_override)) then select_first([load_data_maxretries_override])
+                                       else if (effective_load_data_batch_size < 12) then 3
+                                            else effective_load_data_batch_size / 4
+
```

Comment:
nit: if you are already changing things---I would appreciate a little more prose on why this is being done this way

Reply:
hmm ok, I'll try to come up with something... 😅